common/src/main/java/com/wynntils/screens/maps/IconFilterScreen.java

```diff
@@ -0,0 +1,285 @@
+        return new IconFilterScreen(previousScreen, icons);
+    }
+
+    @Override
+    public void onClose() {
+        McUtils.mc().setScreen(previousScreen);
+    }
+
+    @Override
+    protected void doInit() {
+        dividedWidth = this.width / GRID_DIVISIONS;
+        dividedHeight = this.height / GRID_DIVISIONS;
+        iconButtonSize = (int) (dividedWidth * 4);
+        backgroundX = dividedWidth * 10;
+        backgroundWidth = dividedWidth * 44;
+        backgroundY = dividedHeight * 7;
+        backgroundHeight = dividedHeight * 50;
+
+        scrollButtonHeight = (dividedWidth / Texture.SCROLL_BUTTON.width()) * Texture.SCROLL_BUTTON.height();
+
+        int filterButtonWidth = (int) (dividedWidth * 10);
+
+        // region include/exclude all buttons
+        includeAllButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.iconFilter.includeAll"), (button) -> {
+                            icons.replaceAll((key, value) -> true);
+                            button.active = false;
+                            excludeAllButton.active = true;
+                            populateIcons();
+                        })
+                .pos((int) backgroundX, (int) (dividedHeight * 3))
+                .size(filterButtonWidth, 20)
+                .build();
+
+        includeAllButton.active = icons.containsValue(false);
+
+        this.addRenderableWidget(includeAllButton);
+
+        excludeAllButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.iconFilter.excludeAll"), (button) -> {
+                            icons.replaceAll((key, value) -> false);
+                            button.active = false;
+                            includeAllButton.active = true;
+                            populateIcons();
+                        })
+                .pos((int) backgroundX + filterButtonWidth + 5, (int) (dividedHeight * 3))
+                .size(filterButtonWidth, 20)
+                .build();
+
+        excludeAllButton.active = icons.containsValue(true);
+
+        this.addRenderableWidget(excludeAllButton);
+        // endregion
+
+        // region done button
+        this.addRenderableWidget(
+                new Button.Builder(Component.translatable("screens.wynntils.iconFilter.done"), (button) -> McUtils.mc()
+                                .setScreen(previousScreen))
+                        .pos((int) (dividedWidth * 44), (int) (dividedHeight * 3))
+                        .size(filterButtonWidth, 20)
+                        .build());
+        // endregion
+
+        // The icons that will be displayed
+        usedIcons = new ArrayList<>(icons.keySet());
+
+        populateIcons();
+    }
+
+    @Override
+    public void doRender(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTick) {
+        renderBackground(guiGraphics, mouseX, mouseY, partialTick);
```

Comment:
I think this is left behind from 1.19.4. 1.20.2's `super#render` calls `renderBackground`. Perhaps you don't need this? Also, you might have to call `super#render` first now.